src/main/java/nugget/FindTaskCommand.java

```diff
@@ -0,0 +1,20 @@
+package nugget;
+
+import java.util.ArrayList;
+
+public class FindTaskCommand implements Command {
+    private String searchQuery;
+
+    public FindTaskCommand(String searchQuery) {
+        this.searchQuery = searchQuery;
+    }
+
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws NuggetException {
```

Comment:
I think the abstraction of each command is great. Have you considered giving a more descriptive function name, because in the context of FindTaskCommand now its okay because its the only function. But say in the future if you add to this class another function then it may not be as clear what the function purpose is. But overall awesome work!